be/src/storage/tablet_manager.cpp

```diff
@@ -1168,6 +1177,8 @@ Status TabletManager::_drop_tablet_directly_unlocked(TTabletId tablet_id, Tablet
     TabletSharedPtr dropped_tablet = it->second;
     tablet_map.erase(it);
     _remove_tablet_from_partition(*dropped_tablet);
+    LOG(INFO) << "drop tablet:" << dropped_tablet->tablet_id() << ", stop compaction task";
+    dropped_tablet->stop_compaction();
```

Comment:
compaction task reference unsafe tablet pointer when tablet release since stop_compaction is unblock

Reply:
yes, you are right. I will use shared_ptr for tablet in compaction task to fix it.